### PR TITLE
[LOOP-3375] Onboarding service integration

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-3375
- Update to latest CGMManager, PumpManager, and Service protocols
- Remove special case of pump manager onboarding
- OnboardingManager handles created, but not yet onboarded, view controllers

### CODE REVIEW

**Comment by darinkrauss** [comment]
LW PR for build: https://github.com/tidepool-org/LoopWorkspace/pull/498

File: Loop/Extensions/DeviceDataManager+DeviceStatus.swift [hunk 1]
```diff
@@ -71,7 +71,7 @@ extension DeviceDataManager {
         {
             return .openAppURL(url)
         } else if let cgmManagerUI = (cgmManager as? CGMManagerUI) {
-            return .presentViewController(cgmManagerUI.settingsViewController(for: displayGlucoseUnitObservable, bluetoothProvider: bluetoothProvider, colorPalette: .default))
+            return .presentViewController(cgmManagerUI.settingsViewController(bluetoothProvider: bluetoothProvider, displayGlucoseUnitObservable: displayGlucoseUnitObservable, colorPalette: .default))
```
Comment (nhamming) [inline Loop/Extensions/DeviceDataManager+DeviceStatus.swift:74]: Dexcom plugin will need updating as well, eh?

**Review by nhamming — APPROVED** [review]
As you likely know, plugins still need updating. Besides that LGTM

File: Loop/Extensions/DeviceDataManager+DeviceStatus.swift (hunk 1 above)
Comment (nhamming) [reply Loop/Extensions/DeviceDataManager+DeviceStatus.swift:74]: Just looked at the LW PR and see the plugin updates there.

Comment (darinkrauss) [reply Loop/Extensions/DeviceDataManager+DeviceStatus.swift:74]: Yep, for these wide ranging changes I submit `LoopKit` and `Loop` PRs first to make sure the base code and protocol changes are acceptable. After they are approved, I'll submit the rest of the PRs for review.

**Review by rickpasetto — APPROVED** [review]
LGTM 🧊

**Pull request merged**